tests/testthat/_snaps/verb-fill.md

```diff
@@ -5,16 +5,25 @@
     Output
       <SQL>
       SELECT `id`, `group`, MAX(`n1`) OVER (PARTITION BY `..dbplyr_partion_1`) AS `n1`
-      FROM (SELECT `id`, `group`, `n1`, SUM(CASE WHEN (((`n1`) IS NULL)) THEN (0) WHEN NOT(((`n1`) IS NULL)) THEN (1) END) OVER (ORDER BY -`id` ROWS UNBOUNDED PRECEDING) AS `..dbplyr_partion_1`
-      FROM `df`)
+      FROM (
+        SELECT
+          `id`,
+          `group`,
+          `n1`,
+          SUM(CASE WHEN (((`n1`) IS NULL)) THEN (0) WHEN NOT(((`n1`) IS NULL)) THEN (1) END) OVER (ORDER BY -`id` ROWS UNBOUNDED PRECEDING) AS `..dbplyr_partion_1`
```

Comment:
This looks even worse now 😬

Reply:
Yeah, no changes at this level is going to make this any pretty. But it seems like we could reduce by using an `ELSE`? e.g. `SUM(CASE WHEN `n1` IS NULL THEN 0 ELSE 1 END)` ?

Reply:
Or should we also indent the clauses of a window statement?

```sql
SUM(CASE WHEN (((`n1`) IS NULL)) THEN (0) WHEN NOT(((`n1`) IS NULL)) THEN (1) END) 
  OVER (ORDER BY -`id` ROWS UNBOUNDED PRECEDING) 
  AS `..dbplyr_partion_1`
```

Reply:
That's the translation of `ifelse()`

``` r
library(dbplyr, warn.conflicts = FALSE)
translate_sql(ifelse(is.na(x), 0L, 1L))
#> <SQL> CASE WHEN (((`x`) IS NULL)) THEN (0) WHEN NOT(((`x`) IS NULL)) THEN (1) END
```

<sup>Created on 2021-12-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I think this is because `ELSE` also catches `NULL` (which of course does not matter in this case).

The alternative with `case_when()` is a bit shorter but produces too many lines

``` r
library(dbplyr, warn.conflicts = FALSE)
translate_sql(case_when(is.na(x) ~ 0L, TRUE ~ 1L))
#> <SQL> CASE
#> WHEN (((`x`) IS NULL)) THEN (0)
#> ELSE (1)
#> END
```

<sup>Created on 2021-12-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Indenting the window might make sense. I'll have a look at it.

Reply:
I created a new PR for a nicer translation of the conditionals #738

Reply:
The window clauses currently cannot be indented. They are translated in `sql_build.op_select()` but the level is only known later in the `sql_render()` stage. It might be possible to tackle this with the lazy render pipeline but I'm not sure it's worth it.

Reply:
Ok, makes sense. Not a big deal.